Item Group | test_merge_group_into_leaf | frappe.exceptions.DoesNotExistError: Item Group _Test Item Group B not found

### Similar failed testcases:-
test_merge_groups
test_merge_leaf_into_group
test_merge_leaves
test_move_group_into_another
test_move_group_into_root
test_move_leaf_into_another_group
test_recursion

### Issue Details:
frappe.exceptions.DoesNotExistError: Item Group _Test Item Group B not found

### Traceback:
``` 
ERROR: test_merge_group_into_leaf (erpnext.setup.doctype.item_group.test_item_group.TestItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/setup/doctype/item_group/test_item_group.py", line 229, in test_merge_group_into_leaf
    self.assertRaises(
  File "/usr/lib/python3.10/unittest/case.py", line 738, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/usr/lib/python3.10/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 1452, in rename_doc
    return rename_doc(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/rename_doc.py", line 161, in rename_doc
    old_doc = doc or frappe.get_doc(doctype, old)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 1323, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 85, in get_doc
    return controller(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 126, in __init__
    self.load_from_db()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/document.py", line 172, in load_from_db
    frappe.throw(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 632, in throw
    msgprint(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 597, in msgprint
    _raise_exception()
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 548, in _raise_exception
    raise exc
frappe.exceptions.DoesNotExistError: Item Group _Test Item Group B not found
```

still have tracebacks 
@shubham-DT 
```
======================================================================
ERROR: test_merge_leaves (erpnext.setup.doctype.item_group.test_item_group.TestItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/database/database.py", line 236, in sql
    self._cursor.execute(query, values)
psycopg2.errors.DeadlockDetected: deadlock detected
DETAIL:  Process 33995 waits for ShareLock on transaction 126875; blocked by process 34045.
Process 34045 waits for ShareLock on transaction 126874; blocked by process 33995.
HINT:  See server log for query details.
CONTEXT:  while deleting tuple (32,33) in relation "__global_search"


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/setup/doctype/item_group/test_item_group.py", line 255, in test_merge_leaves
    frappe.rename_doc("Item Group", "_Test Item Group B - 2", "_Test Item Group B - 1", merge=True)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/utils/typing_validations.py", line 31, in wrapper
    return func(*args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 1453, in rename_doc
    return rename_doc(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/rename_doc.py", line 218, in rename_doc
    frappe.delete_doc(doctype, old)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/__init__.py", line 1383, in delete_doc
    return frappe.model.delete_doc.delete_doc(
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/model/delete_doc.py", line 163, in delete_doc
    delete_for_document(doc)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/utils/global_search.py", line 462, in delete_for_document
    frappe.db.delete("__global_search", {"doctype": doc.doctype, "name": doc.name})
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/database/database.py", line 1367, in delete
    return query.run(**kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/query_builder/utils.py", line 87, in execute_query
    result = frappe.db.sql(query, params, *args, **kwargs)  # nosemgrep
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/database/postgres/database.py", line 326, in sql
    return super().sql(modify_query(query), modify_values(values), *args, **kwargs)
  File "/home/frappe/test-postgres/test-bench/apps/frappe/frappe/database/database.py", line 245, in sql
    raise frappe.QueryDeadlockError(e) from e
frappe.exceptions.QueryDeadlockError: deadlock detected
DETAIL:  Process 33995 waits for ShareLock on transaction 126875; blocked by process 34045.
Process 34045 waits for ShareLock on transaction 126874; blocked by process 33995.
HINT:  See server log for query details.
CONTEXT:  while deleting tuple (32,33) in relation "__global_search"


======================================================================
FAIL: test_delete_group (erpnext.setup.doctype.item_group.test_item_group.TestItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/setup/doctype/item_group/test_item_group.py", line 231, in test_delete_group
    self.assertRaises(NestedSetChildExistsError, frappe.delete_doc, "Item Group", "_Test Item Group B")
AssertionError: NestedSetChildExistsError not raised by delete_doc

======================================================================
FAIL: test_move_group_into_another (erpnext.setup.doctype.item_group.test_item_group.TestItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/setup/doctype/item_group/test_item_group.py", line 153, in test_move_group_into_another
    self.assertEqual(old_lft - new_lft, rgt - lft + 1)
AssertionError: 0 != 2

======================================================================
FAIL: test_move_leaf_into_another_group (erpnext.setup.doctype.item_group.test_item_group.TestItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/setup/doctype/item_group/test_item_group.py", line 193, in test_move_leaf_into_another_group
    self.assertEqual(new_rgt - old_rgt, rgt - lft + 1)
AssertionError: 0 != 2

======================================================================
FAIL: test_recursion (erpnext.setup.doctype.item_group.test_item_group.TestItem)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/frappe/test-postgres/test-bench/apps/erpnext/erpnext/setup/doctype/item_group/test_item_group.py", line 121, in test_recursion
    self.assertRaises(NestedSetRecursionError, group_b.save)
AssertionError: NestedSetRecursionError not raised by save
```

--- a/erpnext/setup/doctype/item_group/test_item_group.py
+++ b/erpnext/setup/doctype/item_group/test_item_group.py
@@ -17,51 +17,9 @@ from frappe.utils.nestedset import (
 test_records = frappe.get_test_records("Item Group")
 
 class TestItem(unittest.TestCase):
-
 	def setUp(self):
-		self.insert_dependencies()
-
-		for doc in test_records:
-			if not frappe.db.exists("Item Group", doc["item_group_name"]):
-				frappe.get_doc(doc).insert()
-
-	def insert_dependencies(self):
-		if not frappe.db.exists("Company", "_Test Company"):
-			frappe.get_doc({
-				"doctype": "Company",
-				"company_name": "_Test Company",
-				"abbr": "_TC",
-				"default_currency": "INR"
-			}).insert()
-
-		if not frappe.db.exists("Cost Center", "_Test Cost Center 2 - _TC"):
-			frappe.get_doc({
-				"doctype": "Cost Center",
-				"cost_center_name": "_Test Cost Center 2",
-				"company": "_Test Company",
-				"is_group": 0,
-			}).insert()
-
-		if not frappe.db.exists("Warehouse", "_Test Warehouse - _TC"):
-			frappe.get_doc({
-				"doctype": "Warehouse",
-				"warehouse_name": "_Test Warehouse",
-				"company": "_Test Company",
-				"is_group": 0,
-			}).insert()
-
-		if not frappe.db.exists("Item Tax Template", "_Test Account Excise Duty @ 10 - _TC"):
-			frappe.get_doc({
-				"doctype": "Item Tax Template",
-				"title": "_Test Account Excise Duty @ 10",
-				"company": "_Test Company",
-				"taxes": [
-					{
-						"tax_type": "Excise Duty",
-						"tax_rate": 10
-					}
-				]
-			}).insert()
+		frappe.set_user("Administrator")
+		self.create_test_item_groups()
 
 	def test_basic_tree(self, records=None):
 		min_lft = 1
@@ -280,3 +238,119 @@ class TestItem(unittest.TestCase):
 			"_Test Item Group B - 3",
 			merge=True,
 		)
+
+
+	def create_test_item_groups(self):
+		data = [
+			{
+				"doctype": "Item Group",
+				"is_group": 0,
+				"item_group_name": "_Test Item Group",
+				"parent_item_group": "All Item Groups",
+				"item_group_defaults": [{
+					"company": "_Test Company",
+					"buying_cost_center": "_Test Cost Center 2 - _TC",
+					"selling_cost_center": "_Test Cost Center 2 - _TC",
+					"default_warehouse": "_Test Warehouse - _TC"
+				}]
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 0,
+				"item_group_name": "_Test Item Group Desktops",
+				"parent_item_group": "All Item Groups"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group A",
+				"parent_item_group": "All Item Groups"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group B",
+				"parent_item_group": "All Item Groups"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group B - 1",
+				"parent_item_group": "_Test Item Group B"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group B - 2",
+				"parent_item_group": "_Test Item Group B"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 0,
+				"item_group_name": "_Test Item Group B - 3",
+				"parent_item_group": "_Test Item Group B"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group C",
+				"parent_item_group": "All Item Groups"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group C - 1",
+				"parent_item_group": "_Test Item Group C"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group C - 2",
+				"parent_item_group": "_Test Item Group C"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group D",
+				"parent_item_group": "All Item Groups"
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 1,
+				"item_group_name": "_Test Item Group Tax Parent",
+				"parent_item_group": "All Item Groups",
+				"taxes": [
+					{
+						"doctype": "Item Tax",
+						"parentfield": "taxes",
+						"item_tax_template": "_Test Account Excise Duty @ 10 - _TC",
+						"tax_category": ""
+					},
+					{
+						"doctype": "Item Tax",
+						"parentfield": "taxes",
+						"item_tax_template": "_Test Account Excise Duty @ 12 - _TC",
+						"tax_category": "_Test Tax Category 1"
+					}
+				]
+			},
+			{
+				"doctype": "Item Group",
+				"is_group": 0,
+				"item_group_name": "_Test Item Group Tax Child Override",
+				"parent_item_group": "_Test Item Group Tax Parent",
+				"taxes": [
+					{
+						"doctype": "Item Tax",
+						"parentfield": "taxes",
+						"item_tax_template": "_Test Account Excise Duty @ 15 - _TC",
+						"tax_category": ""
+					}
+				]
+			}
+		]
+
+		for d in data:
+			if not frappe.db.exists("Item Group", d["item_group_name"]):
+				doc = frappe.get_doc(d)
+				doc.insert()

--- a/erpnext/setup/doctype/item_group/test_item_group.py
+++ b/erpnext/setup/doctype/item_group/test_item_group.py
@@ -4,6 +4,7 @@
 
 import unittest
 
+from frappe.test_runner import make_test_records_for_doctype
 import frappe
 from frappe.utils.nestedset import (
 	NestedSetChildExistsError,
@@ -18,8 +19,8 @@ test_records = frappe.get_test_records("Item Group")
 
 class TestItem(unittest.TestCase):
 	def setUp(self):
-		frappe.set_user("Administrator")
-		self.create_test_item_groups()
+		super().setUp()
+		make_test_records_for_doctype("Item Group", force=True)
 
 	def test_basic_tree(self, records=None):
 		min_lft = 1
@@ -238,119 +239,3 @@ class TestItem(unittest.TestCase):
 			"_Test Item Group B - 3",
 			merge=True,
 		)
-
-
-	def create_test_item_groups(self):
-		data = [
-			{
-				"doctype": "Item Group",
-				"is_group": 0,
-				"item_group_name": "_Test Item Group",
-				"parent_item_group": "All Item Groups",
-				"item_group_defaults": [{
-					"company": "_Test Company",
-					"buying_cost_center": "_Test Cost Center 2 - _TC",
-					"selling_cost_center": "_Test Cost Center 2 - _TC",
-					"default_warehouse": "_Test Warehouse - _TC"
-				}]
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 0,
-				"item_group_name": "_Test Item Group Desktops",
-				"parent_item_group": "All Item Groups"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group A",
-				"parent_item_group": "All Item Groups"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group B",
-				"parent_item_group": "All Item Groups"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group B - 1",
-				"parent_item_group": "_Test Item Group B"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group B - 2",
-				"parent_item_group": "_Test Item Group B"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 0,
-				"item_group_name": "_Test Item Group B - 3",
-				"parent_item_group": "_Test Item Group B"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group C",
-				"parent_item_group": "All Item Groups"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group C - 1",
-				"parent_item_group": "_Test Item Group C"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group C - 2",
-				"parent_item_group": "_Test Item Group C"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group D",
-				"parent_item_group": "All Item Groups"
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 1,
-				"item_group_name": "_Test Item Group Tax Parent",
-				"parent_item_group": "All Item Groups",
-				"taxes": [
-					{
-						"doctype": "Item Tax",
-						"parentfield": "taxes",
-						"item_tax_template": "_Test Account Excise Duty @ 10 - _TC",
-						"tax_category": ""
-					},
-					{
-						"doctype": "Item Tax",
-						"parentfield": "taxes",
-						"item_tax_template": "_Test Account Excise Duty @ 12 - _TC",
-						"tax_category": "_Test Tax Category 1"
-					}
-				]
-			},
-			{
-				"doctype": "Item Group",
-				"is_group": 0,
-				"item_group_name": "_Test Item Group Tax Child Override",
-				"parent_item_group": "_Test Item Group Tax Parent",
-				"taxes": [
-					{
-						"doctype": "Item Tax",
-						"parentfield": "taxes",
-						"item_tax_template": "_Test Account Excise Duty @ 15 - _TC",
-						"tax_category": ""
-					}
-				]
-			}
-		]
-
-		for d in data:
-			if not frappe.db.exists("Item Group", d["item_group_name"]):
-				doc = frappe.get_doc(d)
-				doc.insert()

